### PR TITLE
Update to ParaView v5.1.0

### PR DESCRIPTION
**Pull request by quantumsteve**
Description of work.

This updates Mantid to use ParaView v5.1.0. I updated our CMake version to 3.5 to match what ParaView now requires.  Most of the other changes are quite minor.

[Updated ParaView build instructions](http://www.mantidproject.org/Building_VATES)

**To test:**

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/Jenkins/buildscript [hunk 1]
```diff
@@ -16,7 +16,7 @@ BUILDPKG=true
 ###############################################################################
 # All node currently have PARAVIEW_DIR=4.3.b40280 and PARAVIEW_NEXT_DIR=4.3.1
```
Comment (AndreiSavici) [inline buildconfig/Jenkins/buildscript]: Should this comment be modified?

**Comment by AndreiSavici** [comment]
Works fine on Ubuntu 16.04. I suggest testing it on other platforms, at least RHEL7 and Win7

**Comment by AntonPiccardoSelg** [comment]
Trying to build this on a Windows machine:
*  Built PV5.1.0 using the script for our build servers
*  Checked  out this branch and built it against PV5.0.1
* I get:
```
Severity	Code	Description	Project	File	Line
Error	LNK2019	unresolved external symbol "__declspec(dllimport) public: virtual __cdecl MantidQt::MantidWidgets::DataProcessorWhiteList::~DataProcessorWhiteList(void)" (__imp_??1DataProcessorWhiteList@MantidWidgets@MantidQt@@UEAA@XZ) referenced in function "public: virtual class boost::shared_ptr<class MantidQt::MantidWidgets::GenericDataProcessorPresenter> __cdecl MantidQt::CustomInterfaces::ReflGenericDataProcessorPresenterFactory::create(void)" (?create@ReflGenericDataProcessorPresenterFactory@CustomInterfaces@MantidQt@@UEAA?AV?$shared_ptr@VGenericDataProcessorPresenter@MantidWidgets@MantidQt@@@boost@@XZ)	CustomInterfaces	C:\Builds\Mantid\MantidQt\CustomInterfaces\ReflGenericDataProcessorPresenterFactory.obj	1
``` 
A lot symbol visibility errors related to the `DataProcessorAlgorithm`. @quantumsteve  Did you encounter this?

**Comment by quantumsteve** [comment]
@AntonPiccardoSelg I just built Mantid on Windows 7 and didn't counter the issues you describe above. I am able to start MantidPlot and load a MDHistoWorkspace in the VSI.  

vsi

**Comment by AntonPiccardoSelg** [comment]
Can confirm that the Windows 7 build works, just as @quantumsteve said.

**Comment by martyngigg** [comment]
What's the plan with moving forward here? This is currently holding up merging the symbol visibility changes.

**Comment by quantumsteve** [comment]
I think @AndreiSavici wants a smoke test with RHEL 7. Other than that, I think it is ready to merge.

**Comment by martyngigg** [comment]
OK. Is anyone actually looking at that?

**Comment by quantumsteve** [comment]
Not that I know of :disappointed:

**Comment by martyngigg** [comment]
OK. I have a machine next to me that I can install RHEL7 on. I'll give it a go.

**Comment by martyngigg** [comment]
I've tested this on a RHEL7 box and it all seems functional.

I noticed the same warning as in the Windows version regarding the `vtkTemplateTypeMacro`. Is that something we can get rid of?

I also observed a bug with the latex rendering for the axis labels that is unrelated to these changes. It is described in more detail in #16873.

I'm happy to :shipit:

**Comment by quantumsteve** [comment]
@martyngigg The zero-copy-array support was [reworked substantially](https://github.com/Kitware/VTK/blob/master/Documentation/Doxygen/ChangesVTK-7-1.md#vtkdataarray-refactor-vtkarraydispatch-and-related-tools) and we need to update [vtkMDHistoSignalArray.h](https://github.com/mantidproject/mantid/blob/0fcb1ce35f9dca8dc5c46f742780484a2ee0bd47/Vates/VatesAPI/inc/MantidVatesAPI/vtkMDHWSignalArray.h) to work with the new interface.

**Comment by martyngigg** [comment]
@quantumsteve Thanks. Are we aiming to do that for `3.8`?

**Comment by quantumsteve** [comment]
Yes it should be done, though I haven't looked into the new API enough to have a good feeling for how long it will take.

**Pull request merged**